pyvrp/_ProblemData.pyi

```diff
@@ -3,73 +3,68 @@ from typing import Any, List, Tuple
-            This client's service duration, that is, the amount of time we need
-            to visit the client for.
-        """
-    @property
-    def tw_early(self) -> int:
-        """
-        Returns this client's opening time window.
 
-        Returns
-        -------
-        int
-            Earliest time at which we can visit this client.
-        """
-    @property
-    def tw_late(self) -> int:
-        """
-        Returns this client's closing time window.
-
-        Returns
-        -------
-        int
-            Latest time at which we can visit this client.
-        """
-    @property
-    def x(self) -> int:
-        """
+    Attributes
+    ----------
+    demand
+        The amount this client's demanding.
+    service_duration
+        This client's service duration, that is, the amount of time we need to
+        visit the client for.
+    tw_early
+        Earliest time at which we can visit this client.
+    tw_late
+        Latest time at which we can visit this client.
+    x
         Horizontal coordinate of this client, that is, the 'x' part of the
         client's (x, y) location tuple.
-
-        Returns
-        -------
-        int
-            Client's location.
-        """
-    @property
-    def y(self) -> int:
-        """
+    y
         Vertical coordinate of this client, that is, the 'y' part of the
         client's (x, y) location tuple.
+    """
 
-        Returns
-        -------
-        int
-            Client's location.
-        """
+    demand: int
+    service_duration: int
+    tw_early: int
+    tw_late: int
+    x: int
+    y: int
 
 class ProblemData:
+    """
+    Creates a problem data instance. This instance contains all information
+    needed to solve the vehicle routing problem.
+
+    Parameters
+    ----------
+    coords
+        Array of (x, y) coordinates. The first coordinate at index 0 is assumed
+        to be the depot.
+    demands
+        Array of client demands. The demand at index 0 is assumed to be the
+        depot's demand, and should be zero.
+    nb_vehicles
+        The number of vehicles in this problem instance.
+    vehicle_cap
+        Homogenous vehicle capacity for all vehicles in the problem instance.
+    time_windows
+        Array of (early, late) time windows. The time window at index 0 is
+        assumed to be the depot's time window, and describes the overall time
+        horizon.
+    service_durations
+        Array of service durations, that is, the length of time needed to
+        service a customer upon visiting. The service duration at index 0 is
+        assumed to be the depot's service time, and should be zero.
+    duration_matrix
+        A matrix that gives the travel times between clients (and the depot at
+        index 0). Does not have to be symmetric.
+
+    Notes
+    -----
+    All array data assume that the data at or involving index 0 relates to the
+    depot, and all other indices specify client information.
+    """
+
```

Comment:
In general, should we place the docstring for the constructor at the class level rather than in `__init__`? There are a few more places where this applies I think?

Reply:
Update: I see this follows the standard https://numpydoc.readthedocs.io/en/latest/format.html#documenting-classes, so let's do this everywhere! I think we can handle that in https://github.com/N-Wouda/PyVRP/issues/133.

Reply:
Sounds good to me. I moved this over because the docs won't show `__init__` stuff by default, and we do want to display that kind of information. We'll pick up the rest as we add more docs I think.